src/peft/tuners/lora/layer.py

```diff
@@ -250,13 +250,37 @@ def unmerge(self) -> None:
                 self.weight.data -= self.get_delta_weight(active_adapter)
 
     def get_delta_weight(self, adapter) -> torch.Tensor:
-        return (
-            transpose(
-                self.lora_B[adapter].weight @ self.lora_A[adapter].weight,
-                self.fan_in_fan_out,
-            )
-            * self.scaling[adapter]
-        )
+        """
+        Compute the delta weight for the given adapter. In case users wants to merge the adapter weights that are in
```

Comment:
The sentence starting with `In case users` should be a comment, rather than be in the docstring (implementation detail). Could you also add _why_ this casting is necessary only for float16 on CPU? Same applies for the other `get_delta_weight` docstrings.

Reply:
Makes sense, I have adapted the docstring a bit and added relevant comments as you have suggested!